libraries/api/application/ApiApplication.php

```diff
@@ -47,71 +56,118 @@ public function __construct(\JInput $input = null, Registry $config = null, WebC
 
+		// Setup the router
+		// TODO: Router class not ready
+		// $this->router = new ApiRouter();
 	}
 
+
 	/**
-	 * Dispatch the application
+	 * Method to run the application routines.
 	 *
-	 * @param   string $component The component which is being rendered.
+	 * Most likely you will want to instantiate a controller and execute it, or perform some sort of task directly.
 	 *
 	 * @return  void
 	 *
-	 * @since   3.2
+	 * @since   1.0
 	 */
-	public function dispatch($component = null)
+	protected function doExecute()
 	{
+		// Initialise the application
+		$this->initialiseApp();
 
 	}
 
 	/**
-	 * Method to run the application routines.
-	 *
-	 * Most likely you will want to instantiate a controller and execute it, or perform some sort of task directly.
+	 * Rendering is the process of pushing the document buffers into the template
+	 * placeholders, retrieving data from the document and pushing it into
+	 * the application response buffer.
 	 *
 	 * @return  void
 	 *
-	 * @since   1.0
+	 * @since   3.2
+	 *
+	 * @note    Rendering should be overridden to get rid of the theme files.
 	 */
-	protected function doExecute()
+	protected function render()
 	{
-
+		// Render the document
+		$this->setBody($this->document->render($this->allowCache()));
 	}
 
 	/**
-	 * Initialise the application.
-	 *
-	 * @param   array $options An optional associative array of configuration settings.
+	 * Method to send the application response to the client.  All headers will be sent prior to the main application output data.
 	 *
 	 * @return  void
 	 *
-	 * @since   3.2
+	 * @since   1.0
 	 */
-	protected function initialiseApp($options = array())
+	protected function respond()
 	{
-
+		// Send the content-type header.
+		if (!$this->getResponse()->hasHeader('Content-Type'))
+		{
+			$this->setHeader('Content-Type', $this->mimeType . '; charset=' . $this->charSet);
+		}
+
+		// If the response is set to uncachable, we need to set some appropriate headers so browsers don't cache the response.
+		if (!$this->allowCache())
+		{
+			// Expires in the past.
+			$this->setHeader('Expires', 'Wed, 17 Aug 2005 00:00:00 GMT', true);
+
+			// Always modified.
+			$this->setHeader('Last-Modified', gmdate('D, d M Y H:i:s') . ' GMT', true);
+			$this->setHeader('Cache-Control', 'no-store, no-cache, must-revalidate, post-check=0, pre-check=0', false);
+
+			// HTTP 1.0
+			$this->setHeader('Pragma', 'no-cache');
+		}
+		else
+		{
+			// Expires.
+			if (!$this->getResponse()->hasHeader('Expires'))
+			{
+				$this->setHeader('Expires', gmdate('D, d M Y H:i:s', time() + 900) . ' GMT');
+			}
+
+			// Last modified.
+			if (!$this->getResponse()->hasHeader('Last-Modified') && $this->modifiedDate instanceof \DateTime)
+			{
+				$this->modifiedDate->setTimezone(new \DateTimeZone('UTC'));
+				$this->setHeader('Last-Modified', $this->modifiedDate->format('D, d M Y H:i:s') . ' GMT');
+			}
+		}
+
+		$this->sendHeaders();
+
+		echo json_encode($this->getBody());
```

Comment:
As is, you're right.  But if you want debug mode support and being able to spot back some of the profiler data like we have in the downloads site, you'd need to fully override this method.